src/icp_v3.cc

```diff
@@ -49,39 +48,32 @@ doV3Query(int fd, Ip::Address &from, char *buf, icp_common_t header)
     }
 
     /* The peer is allowed to use this cache */
-    ICP3State *state = new ICP3State (header, icp_request);
-    state->fd = fd;
-    state->from = from;
-    state->url = xstrdup(url);
+    ICP3State state(header, icp_request);
+    state.fd = fd;
+    state.from = from;
+    state.url = xstrdup(url);
 
-    StoreEntry::getPublic (state, url, Http::METHOD_GET);
-}
+    const auto e = storeGetPublic(url, Http::METHOD_GET);
 
-ICP3State::~ICP3State()
-{}
-
-void
-ICP3State::created(StoreEntry *e)
-{
-    debugs(12, 5, "icpHandleIcpV3: OPCODE " << icp_opcode_str[header.opcode]);
     icp_opcode codeToSend;
 
-    if (e && confirmAndPrepHit(*e)) {
+    if (e && state.confirmAndPrepHit(*e)) {
         codeToSend = ICP_HIT;
     } else if (icpGetCommonOpcode() == ICP_ERR)
         codeToSend = ICP_MISS;
     else
         codeToSend = icpGetCommonOpcode();
 
-    icpCreateAndSend(codeToSend, 0, url, header.reqnum, 0, fd, from, al);
+    icpCreateAndSend(codeToSend, 0, url, header.reqnum, 0, fd, from, state.al);
 
-    // TODO: StoreClients must either store/lock or abandon found entries.
-    //if (e)
-    //    e->abandon();
+    if (e)
+        e->abandon(__FUNCTION__);
 
-    delete this;
 }
 
+ICP3State::~ICP3State()
```

Comment:
Since git thinks this code is being changed, please take the opportunity to inline this destructor with "=default".

Reply:
Done at 4149d8f0.